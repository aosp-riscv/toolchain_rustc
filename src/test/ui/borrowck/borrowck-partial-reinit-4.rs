struct Test;

struct Test2(Option<Test>);

impl Drop for Test {
    fn drop(&mut self) {
        println!("dropping!");
    }
}

impl Drop for Test2 {
    fn drop(&mut self) {}
}

fn stuff() {
    let mut x : (Test2, Test2);
    (x.0).0 = Some(Test);
<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
    //~^ ERROR assign of possibly uninitialized variable: `x.0`
=======
    //~^ ERROR assign of possibly-uninitialized variable: `x.0`
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)
}

fn main() {
    stuff()
}
