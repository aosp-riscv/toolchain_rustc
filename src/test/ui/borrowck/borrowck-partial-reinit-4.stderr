<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
error[E0381]: assign of possibly uninitialized variable: `x.0`
=======
error[E0381]: assign of possibly-uninitialized variable: `x.0`
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)
  --> $DIR/borrowck-partial-reinit-4.rs:17:5
   |
LL |     (x.0).0 = Some(Test);
<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
   |     ^^^^^^^ use of possibly uninitialized `x.0`
=======
   |     ^^^^^^^ use of possibly-uninitialized `x.0`
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)

error: aborting due to previous error

For more information about this error, try `rustc --explain E0381`.
